### PR TITLE
Add multiple domains to google SSO

### PR DESCRIPTION
**Pull request by corvuscrypto**
This is an implementation of allowing multiple domains by storing as a comma separated set of values as proposed in Issue #5218. 

Relevant descriptions were updated in the frontend and variables/functions were renamed to reflect the allowance of multiple domains.

Tests were added to test detection of an allowed domain within a setting containing multiple allowed domains.

NOTE: I am not an experienced clojure programmer, but it seems idiomatic to change the test expectations for the domain-checking function to `nil` now that it uses `some` which returns either `nil` or `true`. If you need it to report false, I'll change it real quick to satisfy that. Not sure what's truly good style in the clojure community though.
  
###### Before submitting the PR, please make sure you do the following 
-  [x] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [x] Run the frontend and integration tests with  `yarn && yarn run prettier && yarn run lint && yarn run flow && ./bin/build version uberjar && yarn run test`)
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by camsaul** [comment]
Seems like switching the setting like this would break Google SSO for all existing users until an admin went and set it up again

File: src/metabase/api/session.clj [hunk 1]
```diff
@@ -191,16 +191,21 @@
...
+      (clojure.string/split domains #","))))
+
 (defn- email->domain ^String [email]
   (last (re-find #"^.*@(.*$)" email)))
 
-(defn- email-in-domain? ^Boolean [email domain]
+(defn- email-in-domains? ^Boolean [email domains]
   {:pre [(u/email? email)]}
-  (= (email->domain email) domain))
+  (some #(= (email->domain email) %) (split-domains domains)))
```
Comment (camsaul) [inline src/metabase/api/session.clj]: I would just do this like

```clojure
(contains? (set (str/split domains-str #"\s*,\s*")) (email->domain email))
```

*  `require` `metabase.string :as str` and use that. See https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#require-namespaces-you-use
*  You should probably name the param for this function `domains-str` or something so people don't get confused and think it's a sequence of domains

**Review by camsaul — CHANGES_REQUESTED** [review]
Breaking Google SSO for existing instances is a bad idea 

I think the easiest thing to do would be to just keep the existing setting name (`google-auth-auto-create-accounts-domain`). I don't think it's terribly confusing without the extra `s` at the end as long as the dox continue to explain it can optionally have more than one domain

**Comment by corvuscrypto** [comment]
Updated. Sorry for the delay, didn't think I'd get busy this long, but the changes are made.

Side note: Didn't see string imported in the `metabase` namespace anywhere so I just required the clojure standard string package instead.

**Comment by adrienschuler** [comment]
Hi, what is the status of this PR? 
Or maybe there's an another way to handle multiple domains on google single sign on since?
We are really keen on this feature :)

**Pull request closed without merging**